pandas/_libs/tslibs/period.pyx

```diff
@@ -1186,6 +1186,13 @@ cdef int64_t period_ordinal_to_dt64(int64_t ordinal, int freq) except? -1:
     if ordinal == NPY_NAT:
         return NPY_NAT
 
+    if freq == 11000:
+        # Microsecond, avoid get_date_info to prevent floating point errors
```

Comment:
Is it possible to fix `get_date_info` instead?

Reply:
Maybe.  My first two tries didnt work.

Reply:
so, is it possible to use this path for *all* freqs?

Reply:
definitely not for the lower-frequency freqs because of their to_end attrs e.g. Q-DEC, Q-NOV.